File: app/src/main/java/com/marceljurtz/lifecounter/views/Dicing/DicingPresenter.java
package com.marceljurtz.lifecounter.views.Dicing;

import android.content.SharedPreferences;

import com.marceljurtz.lifecounter.views.About.AboutActivity;
import com.marceljurtz.lifecounter.views.Base.Presenter;
import com.marceljurtz.lifecounter.views.Counter.CounterActivity;
import com.marceljurtz.lifecounter.views.Game.GameActivity;
import com.marceljurtz.lifecounter.models.Color;
import com.marceljurtz.lifecounter.models.Dice;
import com.marceljurtz.lifecounter.enums.MagicColorEnum;
import com.marceljurtz.lifecounter.models.PreferenceManager;
import com.marceljurtz.lifecounter.views.Settings.SettingsActivity;

import java.util.Random;

public class DicingPresenter extends Presenter implements IDicingPresenter {

    private Dice _dice;
    private Random _random;
    private IDicingView _view;

    public DicingPresenter(IDicingView view, SharedPreferences preferences) {
        super(view, preferences);
        _view = view;
        this._random = new Random();
    }

    //region Activity Lifecycle

    @Override
    public void onCreate() {
        _dice = new Dice();
        int num = _random.nextInt(4);
        Color color = new Color(MagicColorEnum.WHITE, _preferences);
        switch(num) {
            case 0:
                color = new Color(MagicColorEnum.BLUE, _preferences);
                break;
            case 1:
                color = new Color(MagicColorEnum.GREEN, _preferences);
                break;
            case 2:
                color = new Color(MagicColorEnum.RED, _preferences);
                break;
            case 3:
                break;
        }
        _view.setBackgroundColor(color);
    }

    //endregion

    @Override
    public void onScreenTap() {
        _view.setDicingText(_dice.throwDice()+"");
    }


    //region NavDrawer Handling

    @Override
    public void onMenuEntrySettingsTap() {
        _view.loadActivity(SettingsActivity.class);
    }

    @Override
    public void onMenuEntryAboutTap() {
        _view.loadActivity(AboutActivity.class);
    }

    @Override
    public void onMenuEntryTwoPlayerTap() {
        PreferenceManager.saveDefaultPlayerAmount(_preferences, 2);
        _view.loadActivity(GameActivity.class);
    }

    @Override
    public void onMenuEntryFourPlayerTap() {
        PreferenceManager.saveDefaultPlayerAmount(_preferences, 4);
        _view.loadActivity(GameActivity.class);
    }

    @Override
    public void onMenuEntryCounterManagerTap() {
        _view.loadActivity(CounterActivity.class);
    }

    //endregion
}

File: app/src/main/java/com/marceljurtz/lifecounter/views/Dicing/DicingPresenter.kt
package com.marceljurtz.lifecounter.views.Dicing

import android.content.SharedPreferences

import com.marceljurtz.lifecounter.views.About.AboutActivity
import com.marceljurtz.lifecounter.views.Base.Presenter
import com.marceljurtz.lifecounter.views.Counter.CounterActivity
import com.marceljurtz.lifecounter.views.Game.GameActivity
import com.marceljurtz.lifecounter.models.Color
import com.marceljurtz.lifecounter.models.Dice
import com.marceljurtz.lifecounter.enums.MagicColorEnum
import com.marceljurtz.lifecounter.models.PreferenceManager
import com.marceljurtz.lifecounter.views.Settings.SettingsActivity

import java.util.Random

class DicingPresenter(_view: IDicingView, preferences: SharedPreferences) : Presenter(_view, preferences), IDicingPresenter {

    private var _dice: Dice? = null
    private val _random: Random

    private val _dicingView: IDicingView = _view;

    init {
        this._random = Random()
    }

    //region Activity Lifecycle

    override fun onCreate() {
        _dice = Dice()
        val num = _random.nextInt(4)
        var color = Color(MagicColorEnum.WHITE, _preferences)
        when (num) {
            0 -> color = Color(MagicColorEnum.BLUE, _preferences)
            1 -> color = Color(MagicColorEnum.GREEN, _preferences)
            2 -> color = Color(MagicColorEnum.RED, _preferences)
            3 -> {
            }
        }
        _dicingView.setBackgroundColor(color)
    }

    //endregion

    override fun onScreenTap() {
        _dicingView.setDicingText(_dice!!.throwDice().toString() + "")
    }


    //region NavDrawer Handling

    override fun onMenuEntrySettingsTap() {
        _view.loadActivity(SettingsActivity::class.java)
    }

    override fun onMenuEntryAboutTap() {
        _view.loadActivity(AboutActivity::class.java)
    }

    override fun onMenuEntryTwoPlayerTap() {
        PreferenceManager.saveDefaultPlayerAmount(_preferences, 2)
        _view.loadActivity(GameActivity::class.java)
    }

    override fun onMenuEntryFourPlayerTap() {
        PreferenceManager.saveDefaultPlayerAmount(_preferences, 4)
        _view.loadActivity(GameActivity::class.java)
    }

    override fun onMenuEntryCounterManagerTap() {
        _view.loadActivity(CounterActivity::class.java)
    }

    //endregion
}
